addons/mail/static/src/components/chat_window_manager/chat_window_manager_tests.js

```diff
@@ -1464,6 +1464,245 @@ QUnit.test('chat window: switch on TAB', async function (assert) {
+        document.activeElement,
+        "after pressing tab on the chat window on the middle, the rightmost chatWindow should have focus"
+    );
+
+    await afterNextRender(() =>
+        triggerEvent(
+            chatWindows.rightMost.querySelector('.o_ComposerTextInput_textarea'),
+            'keydown',
+            { key: 'Tab' },
+        )
+    );
+    assert.strictEqual(
+        chatWindows.leftMost.querySelector('.o_ComposerTextInput_textarea'),
+        document.activeElement,
+        "the leftmost chatWindow should have the focus after pressing tab on the rightmost chat window"
+    );
+});
+
+QUnit.test('chat window: cycle between more than 2 chat windows (RTL)', async function (assert) {
+    /**
+     * InnerWith computation uses following info:
+     * ([mocked] global window width: @see `mail/static/src/utils/test_utils.js:start()` method)
+     * (others: @see mail/static/src/models/chat_window_manager/chat_window_manager.js:visual)
+     *
+     * - chat window width: 325px
+     * - start/end/between gap width: 10px/10px/5px
+     * - hidden menu width: 200px
+     * - global width: 1920px
+     *
+     * Enough space for 3 visible chat windows:
+     *  10 + 325 + 5 + 325 + 5 + 325 + 10 = 1000 < 1920
+     */
+    assert.expect(6);
+
+    Object.assign(this.data.initMessaging, {
+        channel_slots: {
+            channel_channel: [{
+                channel_type: "channel",
+                id: 1,
+                is_minimized: true,
+                is_pinned: true,
+                name: "channel1",
+                state: 'open',
+            }, {
+                channel_type: "channel",
+                id: 2,
+                is_minimized: true,
+                is_pinned: true,
+                name: "channel2",
+                state: 'open',
+            }, {
+                channel_type: "channel",
+                id: 3,
+                is_minimized: true,
+                is_pinned: true,
+                name: "channel3",
+                state: 'open',
+            }],
+        },
+    });
+    await this.start({
+        env: {
+            browser: {
+                innerWidth: 1920,
+            },
+        },
+    });
+    this.env.messaging.locale.update({
+        textDirection: 'rtl'
+    });
+    assert.containsN(
+        document.body,
+        '.o_ChatWindow .o_ComposerTextInput_textarea',
+        3,
+        "initialy, 3 chat windows should be present"
+    );
+    assert.containsNone(
+        document.body,
+        '.o_ChatWindow.o-folded',
+        "all 3 chat windows should be open"
+    );
+
+    let chatWindows = document.querySelectorAll('.o_ChatWindow');
+    chatWindows = Array.from(chatWindows).sort((a, b) => (parseInt(a.style.right) > parseInt(b.style.right)) ? 1 : -1);
+    chatWindows = {
+        leftMost: chatWindows[2],
+        middle: chatWindows[1],
+        rightMost: chatWindows[0],
+    };
```

Comment:
It's weak to rely on `style.right` to determine order, and it's dangerous to reuse node references (i.e. stored in variables) in different steps of a test, because they may be obsolete.

I prefer having explicit order on chat window component (e.g. with `data-order`), which is their order in chat window manager.

In LTR text direction: chat windows are placed from RTL, starting from the right border of global window.
In RTL text direction: chat windows are placed from LTR, starting from the left border of global window.

So test defines text direction in env, and makes assertions on which chat window is focused/is to be focused just based on (logical) order.

If you do so, you should no longer need the variable and sort. You can just assert on a given chat window to have focus.